### ISSUE TITLE
Maturity rating

### ISSUE THREAD

**Issue by rrthomas** [luaposix/luaposix#193]
We should add a maturity rating so we can introduce experimental APIs and change them without upset.

On the whole this should not affect direct wrapping of POSIX functions; and since most of luaposix is just that, it shouldn't have much effect.

Indeed, labelling something experimental may be a sign it should be spun out (e.g. `pipeline`).

**Comment by gvvaughan** [comment]
You'll notice that I recently separated `std.maturity` out for the Lua side in https://github.com/lua-stdlib/lua-stdlib/commit/1701cf3cdf6c3e0f30c3ae1fded882403dda5710 - I'm not sure whether that will be enough to satisfy luaposix's requirements, but I'll experiment with it a little, and maybe ask for you to create a 'stdlib/maturity' project for me to hang it off if things work out?

**Comment by rrthomas** [comment]
I'd be inclined to mark only three things: 1) bindings whose details change for some reason, 2) (non-maturity-related) anything non-POSIX that gets in [FOR A REALLY GOOD REASON!] and 3) bindings that go away because the underlying POSIX API does (that's a different sort of obsolescence, I think).

I think even more now that anything non-POSIX should go in its own library. (A good example is my pipeline code, sigh.)

**Comment by gvvaughan** [comment]
Agreed.  And we're fast closing in on the 1-year anniversary of v33.0.0, which will make room to eject quite a lot of the backwards compatibility cruft supporting changes up to that point.

It sure would be nice to ship an ultra-slim luaposix v34.0.0 in early 2016...
